amplipi/updater/asgi.py

```diff
@@ -95,6 +112,93 @@ class ReleaseInfo(BaseModel):
+  ret = Persist_Logs(persist_logs=journalconf.get("Journal", "Storage", fallback="volatile") == "persistent", auto_off_delay=logconf.get("logging", "auto_off_delay", fallback="14"),)
+  return ret
+
+
+@router.post("/settings/persist_logs")
+def toggle_persist_logs(data: Persist_Logs):
+  """Toggles the option within journald to save logs to memory or storage, and sets the length of time before that setting is reset to volatile"""
+  try:
+    create_logging_ini()  # Just in case
+    state = get_log_persist_state()
+
+    if state.persist_logs != data.persist_logs:
+      journalconf = '/etc/systemd/journald.conf'
+      journaltmp = '/tmp/journald.conf.tmp'
+      journal = configparser.ConfigParser(strict=False, allow_no_value=True)
+      journal.read(journalconf)
+
+      if not journal.has_section("Journal"):
+        journal.add_section('Journal')
+
+      # goal_value is true if you wish to turn persistent logging on and false if you wish to turn it off
+      if data.persist_logs:  # Set persist
+        journal.set('Journal', 'Storage', 'persistent')
+        journal.set('Journal', 'SyncIntervalSec', '30s')
+        journal.set('Journal', 'SystemMaxUse', '64M')
+
+        journal.remove_option('Journal', 'RuntimeMaxUse')
+        journal.remove_option('Journal', 'ForwardToConsole')
+        journal.remove_option('Journal', 'ForwardToWall')
+      else:  # Reset config to default as seen in configure.py
+        journal.set('Journal', 'Storage', 'volatile')
+        journal.set('Journal', 'RuntimeMaxUse', '64M')
+        journal.set('Journal', 'ForwardToConsole', 'no')
+        journal.set('Journal', 'ForwardToWall', 'no')
+
+        journal.remove_option('Journal', 'SyncIntervalSec')
+        journal.remove_option('Journal', 'SystemMaxUse')
+
+      with open(journaltmp, 'w', encoding="utf-8") as conf_file:
+        journal.write(conf_file)
+
+      subprocess.run(['sudo', 'mv', journaltmp, journalconf], check=True)
+      subprocess.run(['sudo', 'systemctl', 'restart', 'systemd-journald'], check=True)
+      logger.info(f"persist_logs set to {data.persist_logs}")
+    else:
+      logger.info("persist_logs unchanged")
+
+    if state.auto_off_delay != data.auto_off_delay:
+      logconf = '/var/log/logging.ini'
+      logtmp = '/tmp/logging.ini.tmp'
+      log = configparser.ConfigParser(strict=False, allow_no_value=True)
+      log.read(logconf)
+      log.set('logging', 'auto_off_delay', f"{data.auto_off_delay}")  # Accept auto_off_delay as an int for type checking, parse to str for configParser validity
+      with open(logtmp, 'w', encoding='utf-8') as file:
+        log.write(file)
+      subprocess.run(['sudo', 'mv', logtmp, logconf], check=True)
+      logger.info(f"auto_off_delay set to {data.auto_off_delay}")
+
+    else:
+      logger.info("auto_off_delay unchanged")
+  except Exception as exc:
+    logger.exception(str(exc))
+    return 500
```

Comment:
I don't think `return 500` is a sensible way to handle things, but it is what we do everywhere else and one of the notes was "do this like everything else" and so I've simply followed the template of the endpoints beneath this